awx/main/tasks.py

```diff
@@ -149,6 +149,85 @@ def handle_setting_changes(self, setting_keys):
             break
 
 
+def _cancel_queues(app, worker_queues, worker_name):
+    removed_queues = []
+    for queue in worker_queues:
+        if queue['name'] in settings.AWX_CELERY_QUEUES_STATIC or \
+                {'alias': queue['name']} in settings.AWX_CELERY_QUEUES_STATIC:
+            continue
+
+        removed_queues.append(queue['name'])
+        app.control.cancel_consumer(queue['name'], reply=True, destination=[worker_name])
+    return removed_queues
+
+
+def _add_queues(app, instance, worker_name):
+    # Add worker to group-named queues
+    queues = []
+    for ig in instance.rampart_groups.all():
+        app.control.add_consumer(ig.name, reply=True, destination=[worker_name])
+        queues.append(ig.name)
+
+    # Add worker to worker-named queue
+    app.control.add_consumer(instance.hostname, reply=True, destination=[worker_name])
+    queues.append(instance.hostname)
+
+    return queues
+
+
+def _update_celery_routes(instance):
```

Comment:
Is this better, at least for `cluster_node_heartbeat` and `purge_old_stdout_files` than having it listed in settings statically?

https://github.com/ansible/awx/blob/devel/installer/image_build/files/settings.py#L35-L36

Reply:
`cluster_node_heartbeat` is a tricky beast. By default, it runs on the default queue, `tower`. However, the static files modify the route to ensure that, when the beat runs, it's submitted to the local nodes queue. This behavior is needed. It effectively turns the celery beat into a cron job.

I reckon we could put it in a static file. I just don't love all the locations that we have to sprinkle the node name. i.e. https://github.com/ansible/tower-packaging/pull/38/files#diff-b95c54dc2d67fae41d4686e5480b626b